internal/services/network/application_gateway_resource.go

```diff
@@ -4460,3 +4582,165 @@ func applicationGatewayCustomizeDiff(ctx context.Context, d *pluginsdk.ResourceD
 
 	return nil
 }
+
+func applicationGatewayHttpListnerHash(v interface{}) int {
+	var buf bytes.Buffer
+
+	if m, ok := v.(map[string]interface{}); ok {
+		buf.WriteString(m["name"].(string))
+		buf.WriteString(m["frontend_ip_configuration_name"].(string))
+		buf.WriteString(m["frontend_port_name"].(string))
+		buf.WriteString(m["protocol"].(string))
+		if v, ok := m["host_name"]; ok {
+			buf.WriteString(v.(string))
+		}
+		if hostNames, ok := m["host_names"]; ok {
+			buf.WriteString(fmt.Sprintf("%s-", hostNames.(*pluginsdk.Set).List()))
+		}
```

Comment:
we'll want to include the values for the items in this?

```suggestion
		if hostNames, ok := m["host_names"]; ok {
			items := make([]string, 0)
			for _, v := range hostNames.(*pluginsdk.Set).List() {
				items = append(items, v.(string))
			}
			buf.WriteString(fmt.Sprintf("%s-", strings.Join(items, ","))
		}
```